server/service/src/invoice/inbound_shipment/update/generate.rs

```diff
@@ -116,7 +116,7 @@ pub fn should_create_batches(invoice: &InvoiceRow, patch: &UpdateInboundShipment
         let invoice_status_index = invoice.status.index();
         let new_invoice_status_index = new_invoice_status.index();
 
-        new_invoice_status_index >= InvoiceRowStatus::Delivered.index()
+        new_invoice_status_index == InvoiceRowStatus::Delivered.index()
```

Comment:
There is an issue to refactor all these to match statements, but this will do for a hotfix XD

Reply:
What if status is moves straight to verified ?

Reply:
Status straight to verified can't be done in the inbound shipment. Have to do in chronological order.
![2024-04-02 11 32 40](https://github.com/msupply-foundation/open-msupply/assets/13145522/e368836b-be48-4eb4-9ab3-8c686b81d858)